ccm_web/server/src/api/api.utils.ts

```diff
@@ -38,7 +39,7 @@ export function parseErrorBody (body: unknown): string {
   return body.errors.map(e => e.message).join(' ')
 }
 
-export function makeResponse<T extends CanvasEntity> (multipleResults: Array<APIErrorData | T>): T[] | APIErrorData {
```

Comment:
I think I suggested this initially, but I was seeing some TypeScript errors when I tried to use it. Because the `T` isn't actually used anywhere, I don't think it's essential that we use `extends` to limit what types are accepted here. Basically, it will be `T` if it's not `APIErrorData`. One cool thing is that by removing this, we can actually use this for calls that return multiple objects, since we can pass, for example, `CanvasCourse[]` as the type parameter, and it all still works!

Reply:
Besides, I am now seeing an error where this function returns `successes` that looks like this:
```
Type 'CanvasEnrollment' is not assignable to type 'T'.
          'CanvasEnrollment' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'CanvasEntity'.ts
```
This sent me down a rabbit hole of SO posts and TypeScript docs, and I'm not entirely sure I can explain it. Perhaps TS can't ensure that the same member of the `CanvasEntity` union type is accepted and returned from the function? The error goes away when you remove the `CanvasEntity` constraint.

Reply:
I appreciate the effort here. We can leave as is now.